features/support/commit_helpers.rb

```diff
@@ -1,13 +1,110 @@
-# Returns the commits in the current directory
-def commits_in_repo keys = [:author, :file_content, :file_name, :message]
-  out = {}
+# Makes finding Git commits easier
+class CommitsFinder
+
+  # commit_attributes - the attributes we want to gather for the commits found
+  #                     The default value is the minimalistic set
+  #                     for checking if commits exist at all
+  #
+  # Examples
+  #
+  #   CommitsFinder.new [:branch, :location, :message, :file_name, :file_content, :author]
+  def initialize commit_attributes = [:message]
+    @commit_attributes = commit_attributes.map(&:upcase)
+
+    # The currently known commits
+    @commits = {}
+  end
+
 
-  existing_branches.each do |branch_name|
-    commits = commits_for_branch branch_name, keys
-    out[branch_name] = commits unless commits.empty?
+  # Adds all commits in the current repo to this CommitsFinder instance
+  def add_commits_in_current_repo
```

Comment:
I'd like to keep the table builder as a generic tool, which provides a variety of means to build workflows for finding commits in a variety of places and output them in a variety of formats. I have created a helper method `commits_in_current_repo` to do what you propose using the CucumberTableBuilder object.